Bump version to 3.1rc4

twiecki opened the pull request:
After merging this we should not merge any other PRs until 3.1 is out.

twiecki committed:
Bump version to 3.1rc4

twiecki committed:
Bump version to 3.1rc4

fonnesbeck commented:
There's one more in `docs/source/conf.py`

fonnesbeck commented:
The CHANGELOG should also be updated.

twiecki commented:
Ah, we have release notes, should probably remove CHANGELOG.

twiecki committed:
Remove deprecated changelog in favor of release-notes.

twiecki committed:
Remove deprecated changelog in favor of release-notes.

twiecki commented:
conf.py is added.

fonnesbeck commented:
Are we happy with the release notes as they are? One easy way to get a run-down of changes is:

    git log --oneline --no-merges v3.0..HEAD

Maybe scan through that and see if there are additions we want to highlight.

junpenglao commented:
I found a bug: #2225

fonnesbeck commented:
I will test it now too.

junpenglao commented:
Is there anything left to be done for release?

aloctavodia commented:
Given that #2294 is a bug fix and not a new feature shouldn't be part of the 3.1 release?

twiecki commented:
Update RELEASE-NOTES if anything changed, otherwise we should go ahead.

junpenglao commented:
We should wait for the bug fix of #2298

ferrine commented:
That's done. Is anything else left?

ferrine commented:
We then again need experimental 3.2 branch for new features with regular rebase on master.

junpenglao commented:
Also remove the 3.1 branch and clean up some other branch.

junpenglao commented:
And also #2258 is still broken.

ferrine commented:
#2290 is broken too

twiecki commented:
@fonnesbeck I think we're ready.

fonnesbeck commented:
OK, will get it rolling today.

fonnesbeck committed:
Merge branch 'master' into bump_v3.1rc4

fonnesbeck commented:
Testing the notebooks now. Can we replace `bayesian_neural_network_advi.ipynb` with `bayesian_neural_network_opvi-advi.ipynb`? The former is using an outdated API for part of it, otherwise it is complete overlap.

fonnesbeck committed:
Fixed API breakage in bayesian_neural_network_opvi-advi notebook

twiecki commented:
@fonnesbeck Yes, change the NB.

fonnesbeck committed:
Minor cleanup of cox model

fonnesbeck committed:
Minor cleanup of cod model

fonnesbeck committed:
Updated bayesian_neural_network_advi with update from OPVI version and removed duplicate notebook

fonnesbeck committed:
Merge branch 'bump_v3.1rc4' of github.com:pymc-devs/pymc3 into bump_v3.1rc4

Commit:
bug fix #2346

missing fix (ref: see exchange in https://github.com/pymc-devs/pymc3/pull/2309/files)

fonnesbeck committed:
Minor tweak to dp mixture notebook

fonnesbeck committed:
Fixes to biased inference notebook

fonnesbeck committed:
Merge branch 'bump_v3.1rc4' of github.com:pymc-devs/pymc3 into bump_v3.1rc4

fonnesbeck committed:
Fixes to dd regression notebook

fonnesbeck committed:
Updated gaussian_process notebook to new VI interface

fonnesbeck committed:
Cleaned up hierarchical GLM notebook

fonnesbeck committed:
Removed interact widget from GP smoothing notebook. Its currently broken.

fonnesbeck committed:
Removed backends section from getting_started

fonnesbeck committed:
Tweaks to LDA and LKJ notebooks

fonnesbeck committed:
Fixed API breakage in model comparison notebook

fonnesbeck committed:
Changed sampling iterations in multilevel modeling notebook

fonnesbeck committed:
Removed GP introduction for 3.1 release. Will get it working again for 3.2 when the GP updates will be added.

fonnesbeck committed:
Language edit for tips and tricks notebook

fonnesbeck committed:
Minor tweak to rugby analytics notebook

fonnesbeck committed:
Tweaks to stochastic volatility notebook

fonnesbeck committed:
Updated RELEASE-NOTES to include more changes

fonnesbeck committed:
Minor updates to survial example

fonnesbeck committed:
Fixed GP slice sampling failure

fonnesbeck committed:
Reduced number of samples in updating_priors notebook

fonnesbeck committed:
Updated RELEASE-NOTES to reflect Minibatch as the primary interface for minibatch processing

fonnesbeck committed:
Removed SMC notebook for 3.1 release

ferrine commented on RELEASE-NOTES.md:
```diff
@@ -32,7 +32,7 @@
   - [Added Stein-Variational Gradient Descent as well as Amortized SVGD (experimental).](https://github.com/pymc-devs/pymc3/pull/2183)
 
-  - [Added pm.generator() to easily specify mini-batches.](http://pymc-devs.github.io/pymc3/notebooks/bayesian_neural_network_opvi-advi.html#Minibatch-ADVI)
+  - [Added `Minibatch` to easily specify mini-batches.](http://pymc-devs.github.io/pymc3/notebooks/bayesian_neural_network_opvi-advi.html#Minibatch-ADVI)
```
It is not used in this notebook... I think reference to VI intro is more suitable

fonnesbeck committed:
Changed reference notebook for minibatch release note

twiecki committed:
Reorder release notes

twiecki committed:
Fixes to OPVI NB.

twiecki commented:
I rebased this branch against master, make sure to force pull.

twiecki committed:
Fixes to OPVI NB.

twiecki committed:
Add discourse forum.

twiecki commented:
I wonder if we should just go to 3.1 immediately. We've been testing this a lot and had several successful RCs already.

fonnesbeck commented:
Yes, I was just going to use this RC to ensure that it installs and tests pass, after which I will immediately tag 3.1.

fonnesbeck commented:
Will do it in a separate PR.

fonnesbeck committed:
Merge branch 'bump_v3.1rc4' of github.com:pymc-devs/pymc3 into bump_v3.1rc4

junpenglao commented:
What about the new docs http://docs.pymc.io/ ?

twiecki commented:
@junpenglao It's functional but all the links still go the old site, so we would need to figure out a redirect before. I think we better leave that for the next release.

junpenglao commented:
@twiecki Cool. So I guess we are ready?

fonnesbeck commented:
OK, will get the official 3.1 going then.

twiecki commented:
@fonnesbeck trying to upload 3.1rc4 to pypi but airport wifi is not cooperating. Let's do that, everyone install on a clean base and run tests and then go for 3.1.

twiecki commented:
I think so! @fonnesbeck?

On Jun 25, 2017 3:39 PM, "Junpeng Lao" <notifications@github.com> wrote:

> @twiecki <https://github.com/twiecki> Cool. So I guess we are ready?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pymc-devs/pymc3/pull/2223#issuecomment-310903207>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AApJmCGjIFoqYfWESlZie-KlfY7JhdKeks5sHmMigaJpZM4Nnf7C>
> .
>

fonnesbeck commented:
Just pushed a 3.1 PR.
